data/org.freedesktop.impl.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,163 @@
+        @results: Vardict with the results of the call
+
+        Create a global shortcuts session.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>session_id s</term>
+            <listitem><para>
+              The session id. A string representing the created global shortcuts session.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>shortcuts a{sv}</term>
+            <listitem><para>
+              The different shortcut ids that have been registered mapped to user-readable descriptions for the app to
+              present that describes the shortcut.
+            </para></listitem>
+          </varlistentry>
+      </variablelist>
+    -->
+    <method name="CreateSession">
+      <arg type="o" name="handle" direction="in"/>
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="app_id" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="a{sv}" name="results" direction="out"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In3" value="QVariantMap"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out0" value="QVariantMap"/>
+    </method>
+
+    <!--
+        ListShortcuts:
+        @handle: Object path for the #org.freedesktop.impl.portal.Request object representing this call
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @results: Vardict with the results of the call
+
+        List the shortcuts registered in the @p handle session.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>shortcuts a{sv}</term>
+            <listitem><para>
+              The different shortcut ids that have been registered mapped to user-readable descriptions for the app to present that describes the shortcut.
+            </para></listitem>
+          </varlistentry>
+      </variablelist>
+    -->
+    <method name="ListShortcuts">
+      <arg type="o" name="handle" direction="in"/>
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="a{sv}" name="results" direction="out"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out0" value="QVariantMap"/>
+    </method>
+
+    <!--
+        BindShortcuts:
+        @handle: Object path for the #org.freedesktop.impl.portal.Request object representing this call
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @parent_window: Identifier for the application window, see <link linkend="parent_window">Common Conventions</link>
+        @shortcuts: The identifier of the shortcuts we intend to register, empty for all shortcuts
```

Comment:
What happens if one tries to bind shortcuts not part of the `CreateSession` call?

Reply:
It should error out. I am not sure how this is generally communicated, do you know what would be a good example?

Reply:
It can close the session and start a new one with the previous + the added one. This is how I imagined it at least.

Reply:
I see. Re-creating the session is racy; if one triggers a binding right at the time the app decides to recreate the session, it can be lost. Maybe not a big deal since it's very unlikely to happen very often. Then again, could just as well be one session per binding couldn't it?

Reply:
> Re-creating the session is racy; if one triggers a binding right at the time the app decides to recreate the session, it can be lost. Maybe not a big deal since it's very unlikely to happen very often. 

Correct. It shouldn't happen though during general runtime of the app.

> Then again, could just as well be one session per binding couldn't it?

Yes.